src/Microsoft.VisualStudio.Threading/DispatcherExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿/********************************************************
+*                                                        *
+*   © Copyright (C) Microsoft. All rights reserved.      *
+*                                                        *
+*********************************************************/
+
+#if NET45
+
+namespace Microsoft.VisualStudio.Threading
+{
+    using System;
+    using System.Threading;
+    using System.Windows.Threading;
+
+    /// <summary>
+    /// Extension methods for the WPF <see cref="Dispatcher"/> for better
+    /// interop with the <see cref="JoinableTaskFactory"/>.
+    /// </summary>
+    public static class DispatcherExtensions
+    {
+        /// <summary>
+        /// Creates a <see cref="JoinableTaskFactory"/> that schedules work with the specified
+        /// <see cref="Dispatcher"/> and <see cref="DispatcherPriority"/>.
+        /// </summary>
+        /// <param name="joinableTaskFactory">The underlying <see cref="JoinableTaskFactory"/> to use.</param>
+        /// <param name="priority">
+        /// The priority with which to schedule any work on the UI thread,
+        /// when and if <see cref="JoinableTaskFactory.SwitchToMainThreadAsync"/> is called.
+        /// </param>
+        /// <param name="dispatcher">The <see cref="Dispatcher"/> that schedules work on the main thread.</param>
+        /// <returns>A <see cref="JoinableTaskFactory"/> that may be used for scheduling async work with the specified priority.</returns>
+        public static JoinableTaskFactory WithPriority(this JoinableTaskFactory joinableTaskFactory, DispatcherPriority priority, Dispatcher dispatcher)
```

Comment:
I am conflicted on this myself. At one point I considered `Dispatcher` could be an optional parameter, so it needed to be last. But I thought better of that since we could only guess the `Dispatcher` if you called this method from the UI thread, which thread requirement for calling the method is counter to precedent. So assuming it's required, the only reason to leave priority first is that the method is called `WithPriority`. But I'm not confident that that's a good enough argument.